### PR TITLE
Groups, labels and fields for API v2

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: temba/api/v2/views.py [hunk 1]
```diff
@@ -260,6 +267,123 @@ def get_read_explorer(cls):
...
+                    "value_type": "text"
+                },
+                ...
+            ]
+        }
+    """
+    permission = 'contacts.contactfield_api'
+    model = ContactField
+    serializer_class = ContactFieldReadSerializer
+    pagination_class = CreatedOnCursorPagination
```
Comment (rowanseymour) [inline temba/api/v2/views.py]: We don't have indexes for created_on for groups or fields but from testing on a large org it doesn't seem to make any significant difference as the numbers of the these are so small.

File: temba/api/v2/serializers.py [hunk 2]
```diff
@@ -77,6 +78,37 @@ class Meta:
                   'created_on', 'modified_on')
 
 
+class ContactFieldReadSerializer(ReadSerializer):
+    VALUE_TYPES = {
```
Comment (norkans7) [inline temba/api/v2/serializers.py]: these duplicated constants always introduce issues when we add a new type in one place an forget the other may be include a test that checks all objects from VALUE_TYPE_CHOICES are also included here

Comment (nicpottier) [inline temba/api/v2/serializers.py]: Agreed that maybe these should be using constants from values/models.py

File: temba/api/v2/views.py (hunk 1 above)
Comment (nicpottier) [inline temba/api/v2/views.py]: Ya, don't think those will be a concern.

File: temba/values/models.py [hunk 3]
```diff
@@ -35,11 +23,23 @@
...
+    TYPE_STATE = 'S'
+    TYPE_DISTRICT = 'I'
+
+    TYPE_CONFIG = ((TYPE_TEXT, _("Text"), 'text'),
+                   (TYPE_DECIMAL, _("Numeric"), 'numeric'),
+                   (TYPE_DATETIME, _("Date & Time"), 'datetime'),
+                   (TYPE_STATE, _("State"), 'state'),
+                   (TYPE_DISTRICT, _("District"), 'district'))
+
+    TYPE_CHOICES = [(c[0], c[1]) for c in TYPE_CONFIG]
```
Comment (rowanseymour) [inline temba/values/models.py]: If this looks like a sensible approach, we can implement it for other classes

Comment (norkans7) [inline temba/values/models.py]: Yep

Comment (nicpottier) [inline temba/values/models.py]: +1

**Comment by nicpottier** [comment]
merge master in and we can ship this today.

**Pull request merged**